lib/Command/DeleteAccount.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * @author Anna Larch <anna.larch@nextcloud.com>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+namespace OCA\Mail\Command;
+
+use Exception;
+use OCA\Mail\Account;
+use Psr\Log\LoggerInterface;
+use function memory_get_peak_usage;
+use OCA\Mail\Service\AccountService;
+use OCA\Mail\Exception\ClientException;
+use OCA\Mail\Support\ConsoleLoggerDecorator;
+use OCP\AppFramework\Db\DoesNotExistException;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class DeleteAccount extends Command {
+	public const ARGUMENT_ACCOUNT_ID = 'account-id';
+
+	/** @var AccountService */
+	private $accountService;
+
+	/** @var LoggerInterface */
+	private $logger;
+
+	public function __construct(AccountService $service,
+								LoggerInterface $logger) {
+		parent::__construct();
+
+		$this->accountService = $service;
+		$this->logger = $logger;
+	}
+
+	/**
+	 * @return void
+	 */
+	protected function configure() {
+		$this->setName('mail:account:delete');
+		$this->setDescription('Delete an IMAP account');
+		$this->addArgument(self::ARGUMENT_ACCOUNT_ID, InputArgument::REQUIRED);
+	}
+
+	protected function execute(InputInterface $input, OutputInterface $output): int {
+		$consoleLogger = new ConsoleLoggerDecorator(
+			$this->logger,
+			$output
+		);
+
+		$accountId = (int)$input->getArgument(self::ARGUMENT_ACCOUNT_ID);
+
+		try {
+			$account = $this->accountService->findById($accountId);
+		} catch (Exception $e) {
```

Comment:
this will catch a lot of things and not just the `DoesNotExistException` when the account isn't found. Let's do

```php
try {
    ...
} catch (DoesNotExistException $e) {
  $consoleLogger->error("Account does not exist");
  return 1;
}
```

then any other unexpected error will just bubble up and fail hard.

Reply:
Done